src/libraries/System.Private.CoreLib/src/System/String.Searching.cs

```diff
@@ -423,6 +448,104 @@ public int LastIndexOf(char value, int startIndex, int count)
+        /// the specified character.
+        /// </summary>
+        /// <param name="value">The character to seek.</param>
+        /// <param name="startIndex">The search starting position. The search proceeds from <paramref name="startIndex"/> toward the beginning of this instance.</param>
+        /// <param name="comparisonType">One of the enumeration values that specifies the rules for the search.</param>
+        /// <returns>
+        /// The zero-based index position of <paramref name="value"/> from the end of the current instance
+        /// if that character is found, or -1 if it is not.
+        /// </returns>
+        public int LastIndexOf(char value, int startIndex, StringComparison comparisonType)
+        {
+            return LastIndexOf(value, startIndex, startIndex + 1, comparisonType);
+        }
+
+        /// <summary>
+        /// Reports the zero-based index of the last occurrence of the specified character in the current String object.
+        /// Parameters specify the starting search position in the current string, the number of characters in the
+        /// current string to search, and the type of search to use for the specified character.
+        /// </summary>
+        /// <param name="value">The character to seek.</param>
+        /// <param name="startIndex">The search starting position. The search proceeds from <paramref name="startIndex"/> toward the beginning of this instance.</param>
+        /// <param name="count">The number of character positions to examine.</param>
+        /// <param name="comparisonType">One of the enumeration values that specifies the rules for the search.</param>
+        /// <returns>
+        /// The zero-based index position of <paramref name="value"/> from the end of the current instance
+        /// if that character is found, or -1 if it is not.
+        /// </returns>
+        public int LastIndexOf(char value, int startIndex, int count, StringComparison comparisonType)
+        {
+            return comparisonType switch
+            {
+                StringComparison.CurrentCulture or StringComparison.CurrentCultureIgnoreCase => CultureInfo.CurrentCulture.CompareInfo.LastIndexOf(this, value, startIndex, count, GetCaseCompareOfComparisonCulture(comparisonType)),
+                StringComparison.InvariantCulture or StringComparison.InvariantCultureIgnoreCase => CompareInfo.Invariant.LastIndexOf(this, value, startIndex, count, GetCaseCompareOfComparisonCulture(comparisonType)),
+                StringComparison.Ordinal => LastIndexOf(value, startIndex, count),
+                StringComparison.OrdinalIgnoreCase => LastIndexOfCharOrdinalIgnoreCase(value, startIndex, count),
+                _ => throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparisonType)),
+            };
+        }
+
+        private int LastIndexOfCharOrdinalIgnoreCase(char value, int startIndex, int count)
+        {
```

Comment:
You can just call [https://github.com/dotnet/runtime/blob/43bc8e85d16db7a5a152bea86a55b492027bf420/src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.cs#L359](https://github.com/dotnet/runtime/blob/43bc8e85d16db7a5a152bea86a55b492027bf420/src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.cs#L359)